src/data/nat/basic.lean

```diff
@@ -115,7 +115,11 @@ instance nat.comm_cancel_monoid_with_zero : comm_cancel_monoid_with_zero ℕ :=
     λ _ _ _ h1 h2, nat.eq_of_mul_eq_mul_right (nat.pos_of_ne_zero h1) h2,
   .. (infer_instance : comm_monoid_with_zero ℕ) }
 
-attribute [simp] nat.not_lt_zero
+attribute [simp] nat.not_lt_zero nat.succ_ne_zero nat.succ_ne_self
+  nat.zero_ne_one nat.one_ne_zero
+  nat.zero_ne_bit0 nat.zero_ne_bit1 nat.bit0_ne_zero nat.bit1_ne_zero
+  nat.bit0_ne_one nat.bit1_ne_one nat.one_ne_bit0 nat.one_ne_bit1
+  nat.bit0_ne_bit1 nat.bit1_ne_bit0 nat.bit0_ne nat.bit1_ne
```

Comment:
Please don't mark conditional simp lemmas a simp, when we could have better unconditional ones:
```lean
-- no idea if this exists yet
@[simp] lemma nat.bit0_eq_bit0 {m n : ℕ} : bit0 m = bit0 n ↔ m = n := sorry
@[simp] lemma nat.bit1_eq_bit1 {m n : ℕ} : bit1 m = bit1 n ↔ m = n := sorry
```

Reply:
Ah, thank you. I've written the unconditional versions of these lemmas, as well as generalized them to  `[(semi)ring R] [no_zero_divisors R] [char_zero R]` as appropriate.